### PR TITLE
Replace request with superagent

### PR DESCRIPTION
**Pull request by jdebbink**
My primary motivation for the switch to superagent is to be compatible with React Native. The request module brings a lot of stuff that RN does not like (https://github.com/facebook/react-native/issues/3600). Superagent is isomorphic friendly and lighter weight.

Another benefit is this greatly reduces the size of the browser compatible bundle. Using request the bundle size is 1.5 MB unminified. Using superagent reduces the size to 139 KB unminified.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,6 +38,8 @@
     "sinon": "^1.17.0"
   },
   "dependencies": {
-    "request": "^2.63.0"
+    "path": "^0.12.7",
+    "superagent": "^1.4.0",
+    "url": "^0.11.0"
```
Comment (jdebbink) [inline package.json]: Including path and url are affordances when included in a React Native project.

**Comment by jmaxxz** [comment]
This should also fix this issue:
https://community.particle.io/t/error-importing-sparkjs-with-webpack-babel-loader/13730

(which is the exact same issue I am facing)

**Comment by emilyrose** [comment]
Thanks, @jdebbink! :smile_cat:
Great stuff!

**Comment by emilyrose** [comment]
@jdebbink I almost forgot! Would you like to add yourself as a [contributor](https://docs.npmjs.com/files/package.json#people-fields-author-contributors)?

**Comment by emilyrose** [comment]
If not I'm happy to as well, but I think it's better when done by the person who just contributed! :smile_cat: 

thanks again!

**Comment by emilyrose** [comment]
@jdebbink one more thing! (sorry, I was too excited about reducing bundle size for the silly legal things):

Could you please sign our CLA? Thanks!
https://docs.google.com/forms/d/1_2P-vRKGUFg5bmpcKLHO_qNZWGi5HKYnfrrkd-sbZoA/viewform

**Comment by jdebbink** [comment]
@emilyrose Yeah I'll add myself as a contributor. I submitted the CLA thing yesterday.

**Comment by emilyrose** [comment]
thanks @jdebbink, you're awesome!

**Pull request merged**